python-lib/sharepoint_lists.py

```diff
@@ -107,7 +112,24 @@ def flush(self):
                 futures.append(thread_pool_executor.submit(self.parent.client.process_batch, kwargs[offset:len(kwargs)]))
             for future in as_completed(futures):
                 future_result = future.result()  # Necessary to raise any possible future's exception
-        logger.info("All items added")
+        logger.info("{} items written".format(offset+index))
+
+    def upload_rows(self):
+        logger.info("Starting adding rows")
+        index = 0
+        offset = 0
+        kwargs = []
```

Comment:
In case of single thread, working_batch_size == batch_size. 
So why can't just do
```suggestion
        kwargs = []
        for row in self.buffer:
            item = self.build_row_dictionary(row)
            kwargs.append(self.parent.client.get_add_list_item_kwargs(self.list_id, self.list_item_entity_type_full_name, item))
        self.parent.client.process_batch(kwargs)    
        logger.info("{} items written".format(offset+index))
```